app/controllers/reactions_controller.rb

```diff
@@ -48,24 +48,27 @@ def create
       reactable_type: params[:reactable_type],
       category: category,
     ).first
+    result = ""
     if reaction
       current_user.touch
       reaction.destroy
       Notification.send_reaction_notification_without_delay(reaction, reaction.reactable.user)
       Notification.send_reaction_notification_without_delay(reaction, reaction.reactable.organization) if organization_article?(reaction)
-      @result = "destroy"
+      result = "destroy"
     else
-      reaction = Reaction.create!(
+      reaction = Reaction.new(
         user_id: current_user.id,
         reactable_id: params[:reactable_id],
         reactable_type: params[:reactable_type],
         category: category,
       )
-      @result = "create"
+      return render json: { error: reaction.errors.full_messages.join(", "), status: 422 }, status: :unprocessable_entity unless reaction.save
```

Comment:
this is a really obfuscated guard that also contains a side effect (the saving), in addition to being a really long line visually, what about:

```ruby
if reaction.save
else
end
```

?